filament/src/materials/skybox.mat

```diff
@@ -29,26 +29,108 @@ material {
+    float3 linear_srgb_to_oklab(float3 c) 
+    {
+        float l = 0.4122214708 * c.r + 0.5363325363 * c.g + 0.0514459929 * c.b;
+        float m = 0.2119034982 * c.r + 0.6806995451 * c.g + 0.1073969566 * c.b;
+        float s = 0.0883024619 * c.r + 0.2817188376 * c.g + 0.6299787005 * c.b;
+
+        float l_ = pow(l, 1.0/3.0);
+        float m_ = pow(m, 1.0/3.0);
+        float s_ = pow(s, 1.0/3.0);
+
+        return float3(
+            0.2104542553*l_ + 0.7936177850*m_ - 0.0040720468*s_,
+            1.9779984951*l_ - 2.4285922050*m_ + 0.4505937099*s_,
+            0.0259040371*l_ + 0.7827717662*m_ - 0.8086757660*s_
+        );
+    }
+
+    float3 oklab_to_linear_srgb(float3 c) 
+    {
+        float l_ = c.r + 0.3963377774 * c.g + 0.2158037573 * c.b;
+        float m_ = c.r - 0.1055613458 * c.g - 0.0638541728 * c.b;
+        float s_ = c.r - 0.0894841775 * c.g - 1.2914855480 * c.b;
+
+        float l = l_*l_*l_;
+        float m = m_*m_*m_;
+        float s = s_*s_*s_;
+
+        return float3(
+            +4.0767416621 * l - 3.3077115913 * m + 0.2309699292 * s,
+            -1.2684380046 * l + 2.6097574011 * m - 0.3413193965 * s,
+            -0.0041960863 * l - 0.7034186147 * m + 1.7076147010 * s
+        );
+    }
+
+    float computeGradient(float t) {
+        const float DIMMING_FACTOR = 0.6;
+        const float GRAD_START_HIGH = 0.9;
+        const float GRAD_LOCATION = 0.4;
+        const float GRAD_START_LOW = 0.2;
+
+        float grad;
+        if (t >= GRAD_LOCATION) {
+            grad = DIMMING_FACTOR + (t - GRAD_LOCATION) * (1.0 / (GRAD_START_HIGH - GRAD_LOCATION)) * (1.0 - DIMMING_FACTOR);
+        } else {
+            grad = 1.0 - (t - GRAD_START_LOW) * (1.0 / (GRAD_LOCATION - GRAD_START_LOW)) * (1.0 - DIMMING_FACTOR);
+        }
+        return saturate(grad);
+    }
+
+    float3 shaprSkyGradient() {
+        const float SHAPR_THRESH = 0.5;
+
+        float upDirectionMapped = normalize(variable_eyeDirection).y * 0.5 + 0.5;
+        float3 colorRGB = approxInverseSRGB(materialParams.color.rgb);
+        float3 colorOklab = linear_srgb_to_oklab(colorRGB);
+        if (colorOklab.r > SHAPR_THRESH) {
+            // oklab.r is the luminance channel
+            colorOklab.r *= computeGradient(upDirectionMapped);
+        } else {
+            const float SHAPR_DARK_COLORS_BIAS = 2.0;
+            // oklab.r is the luminance channel
+            colorOklab.r *= (SHAPR_DARK_COLORS_BIAS - computeGradient(upDirectionMapped));
+        }
+        return oklab_to_linear_srgb(colorOklab);
+    }
+
     void material(inout MaterialInputs material) {
         prepareMaterial(material);
-        vec4 sky;
-        if (materialParams.constantColor != 0) {
-            sky = materialParams.color;
+        float4 sky;
+        if (materialParams.skyboxType == 0) {
+            sky = float4(approxInverseSRGB(materialParams.color.rgb), materialParams.color.a);
+        } else if (materialParams.skyboxType == 1) {
```

Comment:
Style convention again, are enum cases usually written like this? Wdyt of introducing `static const` values or `#define`-s for cases of `SkyboxType` in this file?

Reply:
Do you mean using `#ifdef` instead of `if` to select between the skybox type and have 3 different compiled skybox shaders (using different values of the define we are branching on)?

Reply:
I meant this:
```cpp
#define SKYBOX_TYPE_COLOR 0
#define SKYBOX_TYPE_GRADIENT 1
#define SKYBOX_TYPE_ENVIRONMENT 2
```

Reply:
Fixed in https://github.com/shapr3d/filament/pull/8/commits/b0524594a0f48e643169c7379b5026f19464c8cf

Reply:
I haven't seen the previous two comments before commiting. We could go for 3 different shaders, which would make it better for the GPU, but need further modifications to Filament's `Skybox` class, and potentially/probably to either `CMakeLists.txt` or another component that handles shader variant generation. I think this is out of scope for the first release, but should be done when optimizing/aggregating material shaders.

Reply:
Actually 2 different shaders, because filling the screen with a solid color should be a rendertarget clear operation.